statescript/executor.go

```diff
@@ -309,7 +322,22 @@ func (l Launcher) ExecuteAll(state, action string, ignoreError bool) error {
 			}
 		}
 
-		log.Debugf("executing script: %s", s.Name())
+		subStatus := fmt.Sprintf("start executing script: %s", s.Name())
+		log.Debugf(subStatus)
+		if report != nil {
+			if err = reportScriptStatus(report, subStatus); err != nil {
+				log.Errorf("statescript: can not send start status to server: %s", err.Error())
+			}
+		}
+		defer func() {
+			if report != nil {
+				if err = reportScriptStatus(report,
+					fmt.Sprintf("finished executing script: %s", s.Name())); err != nil {
+					log.Errorf("statescript: can not send finished status to server: %s", err.Error())
+				}
+			}
+		}()
```

Comment:
You just tested for `report != nil`. I suggest putting the `defer func` block inside the previous if block, and omitting the second `if report != nil` test entirely.

Reply:
Thanks! Fixed.